In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# Any results you write to the current directory are saved as output.

/kaggle/input/titanic/gender_submission.csv
/kaggle/input/titanic/train.csv
/kaggle/input/titanic/test.csv



# XGBoost classification 



# PART 1 : Exploratory Data Analysis (EDA)[^](#3)<a id="3" ></a><br>
With the objective in mind that this kernal aims to explain the workflow of a predictive modelling problem for begginers, I will try to use simple easy to understand visualizations in the EDA section. Kernals with more advanced EDA sections will be mentioned at the end for you to learn more.

In [2]:
# Python 3 environment comes with many helpful analytics libraries installed
# For example, here's several helpful packages to load in 
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt
import seaborn as sns
import os

In [4]:
# Read data to a pandas data frame
## Load Dataset 
data = pd.read_csv('/kaggle/input/titanic/train.csv')
#test = pd.read_csv('test.csv')
# lets have a look on first few rows
display(data.head())
# Checking shape of our data set
print('Shape of Data : ',data.shape)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Shape of Data :  (891, 12)


* We have 891 data points (rows); each data point has 12 columns.



In [5]:
#checking for null value counts in each column
data.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [ ]:
train = pd.read_csv('/kaggle/input/titanic/train.csv')
test = pd.read_csv('/kaggle/input/titanic/test.csv')
gender_submission = pd.read_csv('/kaggle/input/titanic/gender_submission.csv')

## Dimension of Dataset

In [ ]:
train.shape

In [ ]:
#arrange data into list for labels and list of lists for attributes
xList = []
labels = []
for line in train:
    #split on comma
    row = line.strip().split(",")
    xList.append(row)
sys.stdout.write("Number of Rows of Data = " + str(len(xList)) + '\n')
sys.stdout.write("Number of Columns of Data = " + str(len(xList[1])))

In [ ]:
print(f'{train.shape[0]} samples and {train.shape[1]} features in the training dataset.')

Dataset comprises of 891 observations and 12 characteristics

Out of which one is dependent variable and rest 11 are independent variables

# Peek at the Data

In [ ]:
train.head()

In [ ]:
train.tail()

In [ ]:
train.info()

In [ ]:
train.columns

It is also a good practice to know the columns and their corresponding data types,along with finding whether they contain null values or not

Data has  float, nteger and object values

There are some variables column have null/missing values: Age, Cabin and Embarked

# Statistical Summary

In [ ]:
train.describe().transpose()

In [ ]:
train.hist(bins=50, figsize=(30,20));

The median and the mean both measure central tendency. But unusual values, called outliers, affect the median less than they affect the mean. When you have unusual values, you can compare the mean and the median to decide which is the better measure to use. If your data are symmetric, the mean and median are similar.

In these results, the mean Survived that is required to remove a toothpaste cap is 0.383838, and the median Survived is 0. The data appear to be skewed to the right, which explains why the mean is greater than the median.

For Pclass, the mean is 2.308642 and the median Pclass is 3.0000 that means the data appear to be skewed to the left. 